src/__tests__/nodeSuites/ip-addresses-setting.spec.js

```diff
@@ -52,44 +54,52 @@ const configWithIPAddressesDefault = {
 
 export default function ipAddressesSettingAssertions(mock, assert) {
 
-  let splitioEnabled = SplitFactory(configWithIPAddressesEnabled);
-  let clientEnabled = splitioEnabled.client();
-  const settingsEnabled = SettingsFactory(configWithIPAddressesEnabled);
-
-  mock.onPost(settingsEnabled.url('/events/bulk')).replyOnce(req => {
-    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, true, `Request must include ${HEADER_SPLITSDKMACHINEIP} header.`);
-    assert.equal(HEADER_SPLITSDKMACHINENAME in req.headers, true, `Request must include ${HEADER_SPLITSDKMACHINENAME} header.`);
-    clientEnabled.destroy();
-    return [200];
-  });
-
-  clientEnabled.track('nicolas@split.io', 'sometraffictype', 'someEvent', 10);
+  const finish = (function*() {
+    const clients = [];
+    let count = 1;
+    do{
+      const currentYield = yield;
+      if(currentYield) {
+        clients.push(currentYield);
+        count--;
+      }else
+        count++;
+    }while(count);
+    clients.forEach(client => client.destroy());
+    assert.end();
+  })();
 
-  let splitioDisabled = SplitFactory(configWithIPAddressesDisabled);
-  let clientDisabled = splitioDisabled.client();
-  const settingsDisabled = SettingsFactory(configWithIPAddressesDisabled);
+  function assertHeaders(IPAddressesEnabled, req){
+    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, IPAddressesEnabled, `Request must include ${HEADER_SPLITSDKMACHINEIP} header if IPAddressesEnabled is true.`);
+    assert.equal(HEADER_SPLITSDKMACHINENAME in req.headers, IPAddressesEnabled, `Request must include ${HEADER_SPLITSDKMACHINENAME} header if IPAddressesEnabled is true.`);
+    if(IPAddressesEnabled){
+      assert.equal(req.headers[HEADER_SPLITSDKMACHINEIP], HEADER_SPLITSDKMACHINEIP_VALUE, `${HEADER_SPLITSDKMACHINEIP} header must be equal to the machine ip.`);
+      assert.equal(req.headers[HEADER_SPLITSDKMACHINENAME], HEADER_SPLITSDKMACHINENAME_VALUE, `${HEADER_SPLITSDKMACHINENAME} header must be equal to the machine name.`);
+    }
+  }
 
-  mock.onPost(settingsDisabled.url('/events/bulk')).replyOnce(req => {
-    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, false, `Request must not include ${HEADER_SPLITSDKMACHINEIP} header.`);
-    assert.equal(HEADER_SPLITSDKMACHINENAME in req.headers, false, `Request must not include ${HEADER_SPLITSDKMACHINENAME} header.`);
-    clientDisabled.destroy();
-    return [200];
-  });
+  function mockAndAssertIPAddressesEnabled(config, endpointToMock, clientAction){
+    finish.next();
 
-  clientDisabled.track('nicolas@split.io', 'sometraffictype', 'someEvent', 10);
+    const splitio = SplitFactory(config);
+    const client = splitio.client();
+    const settings = SettingsFactory(config);
+  
+    mock.onPost(settings.url(endpointToMock)).replyOnce(req => {
```

Comment:
We should try to mock all endpoints where we post data, for events, impressions and both for telemetry (counts and times).

You can either use a generator function or keep track of these with a counter of some sorts, as long as you do it inside the `mockAndAssertIPAddressesEnabled` function.

Reply:
Done